### PR TITLE
Fix passing non-existent profile to scarb

### PR DESCRIPTION
**Pull request by THenry14**
Closes https://github.com/foundry-rs/starknet-foundry/issues/1693

## Introduced changes

- fixes a bug, where we allowed passing non existent profile to scarb, which resulted with missing artifacts error

## Checklist

- [X] Linked relevant issue
- [X] Updated relevant documentation
- [X] Added relevant tests
- [X] Performed self-review of the code
- [X] Added changes to `CHANGELOG.md`

### CODE REVIEW

**Review by piotmag769 — APPROVED** [review]
Do we want to release a hotfix with that? It is kinda an annoying bug tbh that makes declare fail for all profiles expect dev and default

File: crates/sncast/src/helpers/scarb_utils.rs [hunk 1]
```diff
@@ -154,7 +154,10 @@ pub fn build_and_load_artifacts(
     build(package, config).map_err(|e| anyhow!(format!("Failed to build using scarb; {e}")))?;
 
     let metadata = get_scarb_metadata_with_deps(&config.scarb_toml_path)?;
-    get_contracts_map(&metadata, &package.id, Some(&config.profile))
+    match &metadata.profiles.contains(&config.profile) {
```
Comment (ddoktorski) [inline crates/sncast/src/helpers/scarb_utils.rs:157]: How about displaying a warning message regarding the Scarb profile being used in the following scenarios?
1. No profile has been provided 
2. The provided profile does not exist

Comment (piotmag769) [reply crates/sncast/src/helpers/scarb_utils.rs:157]: I think a message informing what scarb profile is used is a good thing in general

**Comment by THenry14** [comment]
> Do we want to release a hotfix with that? It is kinda an annoying bug tbh that makes declare fail for all profiles expect dev and default

yeah, I think so. release would work as well, but I think most people do not have this tightly coupled

File: crates/sncast/src/helpers/scarb_utils.rs (hunk 1 above)
Comment (THenry14) [reply crates/sncast/src/helpers/scarb_utils.rs:157]: Yeah, makes sense, I'd do it only for profiles that do not exist in scarb (as I don't think we should inform when profile is not set, scarb does not inform about its default and neither should we imo).

Something like: "Profile <name> does not exist in scarb, using the default "dev" profile" ?

Comment (ddoktorski) [reply crates/sncast/src/helpers/scarb_utils.rs:157]: Sounds good

Comment (THenry14) [reply crates/sncast/src/helpers/scarb_utils.rs]: added warn message in https://github.com/foundry-rs/starknet-foundry/pull/1695/commits/446fbb3b942438cfbcfe4c972ba27074a85c5905

**Review by piotmag769 — APPROVED** [review]

**Pull request merged**